### ISSUE TITLE
Feature request: Don't pair

### ISSUE THREAD

**Issue by vivarose** [tdimiduk/groupeng#5]
I would like a feature where I can name people not to pair with a particular person. In my use case, I have students working in pairs, and I have also assigned anonymous peer reviewers. I would like to assign the students to new partners, but I don't want them to be partnered with someone who has been their peer reviewer.

**Comment by tdimiduk** [comment]
That you can actually do with a distribute rule. Make a column and give each pair (or larger group) you don't want together a unique key (number, letter, word, ...) and assign a distribute rule to the column. That will then cause all the students from, for example reviewer 1 to be assigned to different groups (consider making the rule a high priority one, it shouldn't interfere with other rules much, but other rules could interfere with it)

**Comment by vivarose** [comment]
That is indeed what I did. However, it was cumbersome. I needed seven columns, named avoid1 to avoid7, to fully specify what I had previously specified in just two columns, and it's hard for me to estimate how much time it took me to manually translate the two columns into the seven columns because it took me multiple sessions.

**Comment by vivarose** [comment]
No, I have an interlocking web of students who must avoid each other.

Student A was an anonymous reviewer for students B and C, so needs to avoid students B and C.
Student B was an anonymous reviewer for students D and E, so needs to avoid them.
Student C was an anonymous reviewer for students A, E, and F (three students because A and F wrote a report together), so needs to avoid them.

In the end, pairing B and C is good. Neither one has peer reviewed the other. So I can't say that A, B, C are one distribute group, or else I miss that possible pairing. I have to make A and B a distribute group and then I need a new column so I can make A and C a distribute group.

**Comment by tdimiduk** [comment]
Ok, yeah, that looks like it would need a new rule. I'll think about how to
to that, if you can provide an example file with the info in a form you
would like to provide it in, that would be helpful

**Comment by vivarose** [comment]
(I sent Tom the files by email.)
